test/core/transport/chaotic_good/client_transport_test.cc

```diff
@@ -0,0 +1,390 @@
+  ClientMetadataHandle md;
+  auto args = CallArgs{
+      std::move(md), ClientInitialMetadataOutstandingToken::Empty(), nullptr,
+      nullptr,       &pipe_client_to_server_messages_.receiver,      nullptr};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  EXPECT_CALL(control_endpoint_, Write).Times(3).WillRepeatedly(Return(true));
+  EXPECT_CALL(data_endpoint_, Write).Times(3).WillRepeatedly(Return(true));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: send messages into the pipe, and receive from the
+          // pipe.
+          Join(Seq(pipe_client_to_server_messages_.sender.Push(
+                       std::move(message)),
+                   pipe_client_to_server_messages_.sender.Push(
+                       std::move(message)),
+                   pipe_client_to_server_messages_.sender.Push(
+                       std::move(message)),
+                   [this] {
+                     this->pipe_client_to_server_messages_.sender.Close();
+                     return absl::OkStatus();
+                   }),
+               client_transport_.AddStream(std::move(args))),
+          // Once complete, verify successful sending and the received value.
+          [](const std::tuple<absl::Status, absl::Status>& ret) {
+            EXPECT_TRUE(std::get<0>(ret).ok());
+            EXPECT_TRUE(std::get<1>(ret).ok());
+            return absl::OkStatus();
+          }),
+      InlineWakeupScheduler(),
+      [&on_done](absl::Status status) { on_done.Call(std::move(status)); });
+  // Wait until ClientTransport's internal activities to finish.
+  event_engine_->TickUntilIdle();
+  event_engine_->UnsetGlobalHooks();
+}
+
+TEST_F(ClientTransportTest, AddMultipleStreams) {
+  SliceBuffer buffer;
+  buffer.Append(Slice::FromCopiedString("test add stream."));
+  auto message = arena_->MakePooled<Message>(std::move(buffer), 0);
+  ClientMetadataHandle md;
+  auto first_stream_args = CallArgs{
+      std::move(md), ClientInitialMetadataOutstandingToken::Empty(), nullptr,
+      nullptr,       &pipe_client_to_server_messages_.receiver,      nullptr};
+  auto second_stream_args = CallArgs{
+      std::move(md), ClientInitialMetadataOutstandingToken::Empty(),   nullptr,
+      nullptr,       &pipe_client_to_server_messages_second_.receiver, nullptr};
+  StrictMock<MockFunction<void(absl::Status)>> on_done;
+  EXPECT_CALL(on_done, Call(absl::OkStatus()));
+  EXPECT_CALL(control_endpoint_, Write).Times(2).WillRepeatedly(Return(true));
+  EXPECT_CALL(data_endpoint_, Write).Times(2).WillRepeatedly(Return(true));
+  auto activity = MakeActivity(
+      Seq(
+          // Concurrently: send messages into the pipe, and receive from the
+          // pipe.
+          Join(
+              // Send message to first stream pipe.
+              Seq(pipe_client_to_server_messages_.sender.Push(
+                      std::move(message)),
```

Comment:
Same comment as above. I think a new message should be allocated for each std::move

Reply:
Done, thanks for catching this.